source/plugins/go/src/oms.go

```diff
@@ -1177,51 +1179,82 @@ func PostInputPluginRecords(inputPluginRecords []map[interface{}]interface{}) in
 			msgPackEntries = append(msgPackEntries, msgPackEntry)
 		}
 
-		if !IsWindows && len(msgPackEntries) > 0 { //for linux, mdsd route
-			Log("Info::mdsd:: using mdsdsource name for input plugin records: %s", tag)
+		if (!IsWindows || (IsWindows && IsAADMSIAuthMode)) && len(msgPackEntries) > 0 {
```

Comment:
can we simplify this? we can check for len(msgPackEntries) seperately as a precheck

Reply:
Do you just want to me to create another level of if?

Reply:
something like
if len(msgpackentries) == 0 continue

if !Windows || Windows && MSI {
	...
}